packages/flutter/lib/src/material/chip.dart

```diff
@@ -278,6 +278,19 @@ abstract interface class ChipAttributes {
   /// ** See code in examples/api/lib/material/chip/chip_attributes.chip_animation_style.0.dart **
   /// {@end-tool}
   ChipAnimationStyle? get chipAnimationStyle;
+
+  /// The cursor for a mouse pointer when it enters or is hovering over the
+  /// widget.
+  ///
+  /// If [mouseCursor] is a [WidgetStateMouseCursor],
```

Comment:
Can the mouse cursor in this PR really be resolved in WidgetStates? If it can, a test for that behavior should be added. Otherwise, this documentation block would be misleading.

Reply:
@victorsanni, I added some tests in [test: Add test for mouse cursor being a WidgetStateMouseCursor](https://github.com/flutter/flutter/pull/159422/commits/d7600ce0ef1df14b31e0420a674490583354d0f0).

However I'm facing some issues.

The mouse cursor is resolved in the `InkResponse` which is also the one modifying the `WidgetStatesController statesController` 

https://github.com/flutter/flutter/blob/4a4d7a7dac8aab7c3b27d44352d994cfea7b2925/packages/flutter/lib/src/material/ink_well.dart#L1311-L1314

But the `MaterialState.selected` is handled by the `RawChip`. 

https://github.com/flutter/flutter/blob/4a4d7a7dac8aab7c3b27d44352d994cfea7b2925/packages/flutter/lib/src/material/chip.dart#L1006-L1011

[test: Add test for mouse cursor being a WidgetStateMouseCursor](https://github.com/flutter/flutter/pull/159422/commits/d7600ce0ef1df14b31e0420a674490583354d0f0).

Includes an attempt to merge the 2 of them (I haven't updated the doc comment yet) but I'm facing the error:

```
══╡ EXCEPTION CAUGHT BY FOUNDATION LIBRARY ╞════════════════════════════════════════════════════════
The following assertion was thrown while dispatching notifications for WidgetStatesController:
setState() or markNeedsBuild() called during build.
This RawChip widget cannot be marked as needing to build because the framework is already in the
process of building widgets. A widget can be marked as needing to be built during the build phase
only if one of its ancestors is currently building. This exception is allowed because the framework
builds parent widgets before children, which means a dirty descendant will always be built.
Otherwise, the framework might not visit this widget during this build phase.
The widget on which setState() or markNeedsBuild() was called was:
  RawChip
The widget which was currently being built when the offending call was made was:
  InkWell

When the exception was thrown, this was the stack:
#0      Element.markNeedsBuild.<anonymous closure> (package:flutter/src/widgets/framework.dart:5194:9)
#1      Element.markNeedsBuild (package:flutter/src/widgets/framework.dart:5206:6)
#2      State.setState (package:flutter/src/widgets/framework.dart:1224:15)
#3      MaterialStateMixin.initState.<anonymous closure> (package:flutter/src/material/material_state_mixin.dart:139:7)
#4      ChangeNotifier.notifyListeners (package:flutter/src/foundation/change_notifier.dart:437:24)
#5      WidgetStatesController.update (package:flutter/src/widgets/widget_state.dart:1137:7)
#6      _InkResponseState.initStatesController (package:flutter/src/material/ink_well.dart:883:22)
#7      _InkResponseState.initState (package:flutter/src/material/ink_well.dart:890:5)
#8      StatefulElement._firstBuild (package:flutter/src/widgets/framework.dart:5770:55)
#9      ComponentElement.mount (package:flutter/src/widgets/framework.dart:5615:5)
```


https://github.com/flutter/flutter/blob/4a4d7a7dac8aab7c3b27d44352d994cfea7b2925/packages/flutter/lib/src/material/ink_well.dart#L879-L885


So my question is "How should I tackled this?" 

Should I only support the material state supported "out of the box" by the ink well? Or should I try to merge it with `selected` from the raw chip, and in that case, how do you suggest I could do?

Reply:
In the documentation for [WidgetStatesController](https://api.flutter.dev/flutter/widgets/WidgetStatesController-class.html) :

```
When calling setState in a WidgetStatesController listener, use the SchedulerBinding.addPostFrameCallback to delay the call to setState after the frame has been rendered. It's generally prudent to use the SchedulerBinding.addPostFrameCallback because some of the widgets that depend on WidgetStatesController may call update in their build method. In such cases, listener's that call setState - during the build phase - will cause an error.
```